vars/checkRequestAssignReleaseOwners.groovy

```diff
@@ -0,0 +1,199 @@
+                def awsAccessKey = env.AWS_ACCESS_KEY_ID
+                def awsSecretKey = env.AWS_SECRET_ACCESS_KEY
+                def awsSessionToken = env.AWS_SESSION_TOKEN
+
+                ReleaseMetricsData releaseMetricsData = new ReleaseMetricsData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, this)
+                ComponentRepoData componentRepoData = new ComponentRepoData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, maintainersIndex, this)
+
+                inputManifestObj.components.each { component ->
+                    def releaseOwner = releaseMetricsData.getReleaseOwners(component.name)
+                    if (releaseOwner == null || releaseOwner.isEmpty()) {
+                        componentsMissingReleaseOwners.add(component.name)
+                        if (action != 'check') {
+                            handleMissingReleaseOwner(component, releaseMetricsData, componentRepoData, action)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if (!componentsMissingReleaseOwners.isEmpty()) {
+        echo("Components missing release owner: ${componentsMissingReleaseOwners}")
+    } else {
+        echo('All components have release owner assigned.')
+    }
+}
+
+/**
+ * Validates input parameters
+ */
+private void validateParameters(Map args) {
+    if (!args.inputManifest || args.inputManifest.isEmpty()) {
+        error "inputManifest parameter is required."
+    } else {
+        args.inputManifest.each { inputManifestFile ->
+            if (!fileExists(inputManifestFile)) {
+                error("Invalid path. Input manifest file does not exist at ${inputManifestFile}")
+            }
+        }
+    }
+
+    String action = args.action
+    println("Action is: ${action}")
+    List<String> validActions = ['check', 'assign', 'request']
+    if (!validActions.contains(action)) {
+        error "Invalid action '${action}'. Valid values: ${validActions.join(', ')}"
+    }
+}
+
+/**
+ * Extracts version from manifest file
+ */
+private String extractVersionFromManifest(String manifestFile) {
+    def yaml = readYaml(file: manifestFile)
+    return yaml.build.version
+}
+
+/**
+ * Handle component with missing release owner
+ */
+private void handleMissingReleaseOwner(def component, ReleaseMetricsData releaseMetricsData,  ComponentRepoData componentRepoData, String action) {
+    String repoName = component.repository.toString().split('/')[-1].replace('.git', '')
+    String releaseIssueUrl = releaseMetricsData.getReleaseIssue(repoName)
+    ArrayList<String> componentMaintainers = componentRepoData.getMaintainers(repoName)
+
+    if (componentMaintainers == null || componentMaintainers.isEmpty()) {
+        echo("No maintainers found for component: ${component.name}. Skipping action.")
+        return
+    }
+
+    List<String> maintainersWithTag = componentMaintainers.collect { "@$it" }
+
+    switch (action) {
```

Comment:
what is the logic to handle different actions?

Reply:
Its mentioned below. Depending on the request or assign. For requesting it will just comment tagging all the maintainers to volunteer one. If assign it will randomly assign one.